Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/canonicalize/message.py

```diff
@@ -69,16 +69,28 @@ def main(inputfile, identifier_template):
     if identifier_index and lines[identifier_index - 1].startswith(identifier_template.format('').split(':')[0]):
         lines[identifier_index - 1] = identifier_template.format(commit)
         identifier_index = identifier_index - 2
-    elif identifier_index and lines[identifier_index - 2].startswith(identifier_template.format('').split(':')[0]):
-        del lines[identifier_index - 2]
-        lines.insert(identifier_index - 1, identifier_template.format(commit))
-        identifier_index = identifier_index - 2
     else:
-        lines.insert(identifier_index, identifier_template.format(commit))
+        for index in [2, 3]:
+            if identifier_index - index > 0 and lines[identifier_index - index].startswith(identifier_template.format('').split(':')[0]):
+                del lines[identifier_index - index]
+                lines.insert(identifier_index - 1, identifier_template.format(commit))
+                identifier_index = identifier_index - 2
+                break
+        else:
+            lines.insert(identifier_index, identifier_template.format(commit))
 
     if lines[identifier_index]:
         lines.insert(identifier_index, '')
 
+    # Turn multiple empty lines at the end of a commit into a single one
```

Comment:
Should we have a test case for this?

Reply:
We do: https://github.com/WebKit/WebKit/pull/241/files#diff-a67d5e0f2139831f939f41a909f52b7e25de86f374eda2faabadec5e67759d93R213